erroranalysis/erroranalysis/_internal/cohort_filter.py

```diff
@@ -32,12 +33,13 @@ def filter_from_cohort(analyzer, filters, composite_filters):
     true_y = analyzer.true_y
     categorical_features = analyzer.categorical_features
     categories = analyzer.categories
-    if not isinstance(df, pd.DataFrame):
-        df = pd.DataFrame(df, columns=feature_names)
-    else:
-        # Note: we make a non-deep copy of the input DataFrame since
-        # we will add columns below
-        df = df.copy()
+    if not is_spark(df):
+        if not isinstance(df, pd.DataFrame):
+            df = pd.DataFrame(df, columns=feature_names)
+        else:
+            # Note: we make a non-deep copy of the input DataFrame since
```

Comment:
a shallow copy? Not sure what non-deep is or if it's the same.

Reply:
oh, yes, that's what you call it I guess.  I will update comment in next PR.